### PR TITLE
#2 added stop message

### PR DESCRIPTION
**Pull request by muhammednagy**
Changes proposed in this pull request:
- Added stop message.

### CODE REVIEW

**Comment by abahmed** [comment]
@muhammednagy Can you mention in message too the port that was listening to?

**Comment by abahmed** [comment]
Fixes #2

**Comment by muhammednagy** [comment]
Will do.

File: gearbox.go [hunk 1]
```diff
@@ -136,7 +136,7 @@ func (gb *gearbox) Start(address string) error {
 	if err != nil {
 		return err
 	}
-
+	gb.httpServer.Name = address
```
Comment (abahmed) [inline gearbox.go:139]: According to https://github.com/valyala/fasthttp/blob/32940977fb67c66fa367b7fc8f59cfa0e4f80be7/server.go#L186, this is not the context to save address

File: gearbox.go [hunk 2]
```diff
@@ -151,6 +151,7 @@ func (gb *gearbox) newHTTPServer() *fasthttp.Server {
 
 // Stop serving
 func (gb *gearbox) Stop() error {
+	fmt.Printf("Service stopped listening to %s", gb.httpServer.Name)
```
Comment (abahmed) [inline gearbox.go:154]: What about
>  gearbox v... stopped listening on %s 
(use gearbox name and version from constants)

File: gearbox.go (hunk 1 above)
Comment (muhammednagy) [reply gearbox.go:139]: oh shit.
okay, will take care of it.
The problem is I spent some time trying to get the address the httpserver is listening to but it's inside a not exported array.
I can see it in the debugger but I can't use it.

File: gearbox.go (hunk 2 above)
Comment (muhammednagy) [reply gearbox.go:154]: Okay.

File: gearbox.go (hunk 1 above)
Comment (abahmed) [reply gearbox.go:139]: You can save it in gearbox struct if you cannot get it from anywhere

Comment (abahmed) [reply gearbox.go:139]: @muhammednagy Don't forget to add comments too

File: gearbox.go [hunk 3]
```diff
@@ -115,6 +115,7 @@ type gearbox struct {
 	httpServer       *fasthttp.Server
 	routingTreeRoot  *routeNode
 	registeredRoutes []*routeInfo
+	address			 string
```
Comment (abahmed) [inline gearbox.go:118]: add comments here

File: gearbox.go [hunk 4]
```diff
@@ -149,8 +150,9 @@ func (gb *gearbox) newHTTPServer() *fasthttp.Server {
 	}
 }
 
-// Stop serving
```
Comment (abahmed) [inline gearbox.go:152]: keep " Stop serving" and  this comment above printing message

File: gearbox.go [hunk 5]
```diff
@@ -149,8 +150,9 @@ func (gb *gearbox) newHTTPServer() *fasthttp.Server {
...
+// Print stopping message and stop serving
 func (gb *gearbox) Stop() error {
+	fmt.Printf("Gearbox V%s stopped listening on %s", Version, gb.address)
```
Comment (abahmed) [inline gearbox.go:155]: What if there is an error is stopping, this message is not in right place.
Check first if shutdown call has an error or on, then print it if there is no error

Comment (muhammednagy) [reply gearbox.go:155]: you are right.
got it.

File: gearbox.go [hunk 6]
```diff
@@ -151,7 +152,15 @@ func (gb *gearbox) newHTTPServer() *fasthttp.Server {
 
 // Stop serving
 func (gb *gearbox) Stop() error {
-	return gb.httpServer.Shutdown()
+	err := gb.httpServer.Shutdown()
+	if err != nil {
+		return err
+	}
+	if gb.address == "" {
```
Comment (abahmed) [inline gearbox.go:159]: Make sure if it's not needed if shutdown will return error if address is not provided. Otherwise LGTM

**Comment by muhammednagy** [comment]
Race detected error didn't occur locally when testing.
Will try to reproduce it locally.
Am using the same command as github.

**Pull request merged**